Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -778,9 +777,22 @@ private bool IsProjectGlobalDeclaration(Declaration candidateDeclaration, Declar
         {
             return candidateDeclaration.ProjectName == scopingDeclaration.ProjectName
                 && !(candidateDeclaration.ParentScopeDeclaration is ClassModuleDeclaration)
-                && (candidateDeclaration.Accessibility == Accessibility.Public
-                    || ((candidateDeclaration.Accessibility == Accessibility.Implicit)
-                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)));
+                && (IsExplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration)
+                || IsImplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration));
+        }
+
+        private bool IsExplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                    && candidateDeclaration.Accessibility == Accessibility.Public;
+        }
+
+        private bool IsImplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                        && candidateDeclaration.Accessibility == Accessibility.Implicit
+                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)
+                        && !candidateDeclaration.IdentifierName.StartsWith("Option ");
```

Comment:
The `DeclarationType` for a *module option* declaration would be `DeclarationType.ModuleOption`. You might want to look at the available [DeclarationType](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Parsing/Symbols/DeclarationType.cs#L8) enum flags to see all possible declaration types; I suspect you're only really interested in a very small subset of values.